src/cutils/mod.rs

```diff
@@ -81,4 +104,14 @@ mod test {
         assert_eq!(strp("\0".as_ptr() as *const libc::c_char), "");
         assert_eq!(strp("hello\0".as_ptr() as *const libc::c_char), "hello");
     }
+
+    #[test]
+    fn test_tty() {
+        use std::fs::File;
+        use std::os::fd::AsRawFd;
+        assert!(!super::safe_isatty(
+            File::open("/bin/sh").unwrap().as_raw_fd()
+        ));
+        assert!(!super::safe_isatty(-837492));
```

Comment:
maybe you could test `STDIN` to be sure the other way around also works?

Reply:
That was removed in f8c0408791f576e37627119c136e9b2f40767c48; that test failed in the CI since in that case there is no terminal (the test works locally).

Reply:
Let's see if 45986fc927a909ac21614bc9b639d9e3867183dc passes the CI